Update Hydrawise for config flow (#28108)
home-assistant/home-assistant.github.io: source/_integrations/hydrawise.markdown

  type: string
scan_interval:
  description: The time interval, in seconds, to poll the Hydrawise cloud.
  required: false
  type: integer
  default: 120
{% endconfiguration %}

To get your API access token log into your [Hydrawise account](https://app.hydrawise.com/config/login) and in the 'My Account Details' section under Account Settings click 'Generate API Key'. Enter that key in your configuration file as `YOUR_API_KEY`.
{% include integrations/config_flow.md %}

## Binary Sensor

Once you have enabled the `hydrawise` integration, add the following to your `configuration.yaml` file:

```yaml
# Example configuration.yaml entry
binary_sensor:
  - platform: hydrawise
```

{% configuration %}
monitored_conditions:
  description: The binary sensors that should be displayed on the frontend.
  required: false
  type: list
  default: All binary sensors are enabled.
  keys:
    is_watering:
      description: The binary sensor is `on` when the zone is actively watering.
    status:
      description: This will indicate `on` when there is a connection to the Hydrawise cloud. It is not an indication of whether the irrigation controller hardware is online.
{% endconfiguration %}

> <div class='note warning'>

Reviewer: Should these warnings really be removed?

Author: I added the first warning back, but the second one isn't relevant since config flow doesn't support modifying the scan interval.

> A binary sensor is created for the controller indicating whether it is currently available via the cloud API.

Reviewer: Create a list like below platforms for consistency?

Author: Done

- [Sensor](#sensor)
- [Switch](#switch)

## Prerequisites

To set up the Hydrawise integration, you must first obtain an API Key.

> 1. Log in at [https://app.hydrawise.com](https://app.hydrawise.com).
> 2. Go to 'Account Details' under the 'My Account' menu (in the upper-right hand corner).
> 3. Under the **Account Settings** section, copy the **API Key**.
>    1. If there is not an API Key present, click the **Generate API Key** button.

Reviewer 2: ```suggestion
1. Login at [https://app.hydrawise.com](https://app.hydrawise.com).
2. Go to **Account Details** under the **My Account** menu (in the upper-right-hand corner).
3. Under the **Account Settings** section, copy the **API Key**.
   1. If no API Key is present, select the **Generate API Key** button.
```
